### ISSUE TITLE
Container removeAtom() does not remove them from Sgroups

### ISSUE THREAD

**Issue by dkatzel-ncats** [cdk/cdk#671]
I've noticed that if a container has sgroups and then you remove atoms from that container that happen to also belong to an sgroup, it won't be removed from the sgroup.  This causes problems when we later want to write this container to a mol file.  When the writer tries to write out the Sgroup, it can't find the atom that was removed and errors out.

I wasn't sure if there is a central place in the code to make this fix or if each `IAtomContainer` implementation has to be changed separately.

**Comment by johnmay** [comment]
Each one needs to be changed, my philosophy with the sgroups was the user needs to manage them when they make operations. If you don't care about them they can just be cleared. We should make them a first class property though (i.e. actual access/mutator method) rather than as just a property.

Feel free to add in the remove logic but note for some cases you may still end up in a nonsensical state. For example in brackets:

```
 - [ - CH2 - CH2 - O - ] -  
```

if i delete one atom should just the atom be removed or the whole bracket? What about if a user removes and atom then adds in another one. For example: 

```
 - [ - CH2 - CH2 - O - ] -  
 - [ - CH2 - CH2 - {deleted} - ] -  
 - [ - CH2 - CH2 - N - ] -  
```

There is no (easy) way to handle such a case and it is up to the user to make sure things stay consistent.

**Comment by dkatzel-ncats** [comment]
I agree they should be a first class property

**Comment by johnmay** [comment]
Fixed - like stereo now any removal results in the Sgroup be removed. The exception is setAtom(i, atom) which does a swap in which case the Sgroup is updated as needed.

**Comment by johnmay** [comment]
Fixed via #821
